extensions/vscode/src/daemon/protocol.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Defines the protocol used by the Dart Frog daemon and custom
+ * type guards to check if an object is a valid message.
+ *
+ * @see {@link https://github.com/VeryGoodOpenSource/dart_frog/blob/main/packages/dart_frog_cli/lib/src/daemon/protocol.dart Dart Frog dart's protocol}
+ */
+
+export interface DaemonMessage {}
```

Comment:
So we really need this interface? can't we just use the abstract class instead?

Reply:
It is useful for multiple reasons:
- Formally matches directly with the [protocol](https://github.com/VeryGoodOpenSource/dart_frog/blob/main/packages/dart_frog_cli/lib/src/daemon/protocol.dart) and its hierarchy. 
- Useful as fallback type when decoding deamon messages from stdout (better than falling back to Object or any)
- Arguably easier when declaring instances of iterables with messages

Reply:
I've now included the method to decode messages in this PR to avoid confusion.